packages/jest-snapshot/src/index.ts

```diff
@@ -122,22 +122,26 @@ const cleanup = (
   hasteFS: HasteFS,
   update: Config.SnapshotUpdateState,
   snapshotResolver: JestSnapshotResolver,
-) => {
+): {
+  filesRemoved: number;
+  filesRemovedList: Array<string>;
+} => {
   const pattern = '\\.' + EXTENSION + '$';
   const files = hasteFS.matchFiles(pattern);
-  const filesRemoved = files.reduce((acc, snapshotFile) => {
+  const list = files.reduce<Array<string>>((acc, snapshotFile) => {
     if (!fileExists(snapshotResolver.resolveTestPath(snapshotFile), hasteFS)) {
       if (update === 'all') {
         fs.unlinkSync(snapshotFile);
       }
-      return acc + 1;
+      return [...acc, snapshotFile];
```

Comment:
Spread inside reduce is bad idea, because performance.
This code works faster
```js
acc.push(snapshotFile);
//...
return acc;
```
Or you can use a `filter` instead of `reduce`.

Reply:
good call, refactored to use `filter`